bin/merge_junit

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+# Initialize variables
+reports_dir=""
+output_file=""
+
+# Function to show usage
+usage() {
+    echo "Usage: $0 -d <reports_dir> -o <output_file>"
+    exit 1
+}
+
+# Parse command-line options
+while getopts "d:o:" opt; do
+    case $opt in
+        d) reports_dir=$OPTARG ;;
+        o) output_file=$OPTARG ;;
+        ?) usage ;;
+    esac
+done
+
+# Check if both arguments were provided
+if [ -z "$reports_dir" ] || [ -z "$output_file" ]; then
+    usage
+fi
+
+touch "$output_file"
+{
+  echo '<?xml version="1.0" encoding="UTF-8"?>'
+  # Write XML header to the output file
+  echo '<testsuites>'
+
+  # Merge the content of all input JUnit files in the directory.
+  # (Note that in the case of Unit Tests, the JUnit XML files produced by Gradle
+  # don't have a parent `<testsuites>` root tag, so there's no need to try and remove it)
```

Comment:
What do you suggest? I think this is a standard behavior of JUnit working with Android Gradle Plugin - there's nothing specific in WooCommerce Android.

Reply:
My only fear is that someone would be tempted to use this `merge_junit_report` utility also on JUnit reports that don't come from Unit Tests created by Android Gradle Plugin, like JUnit reports created by fastlane, or by other wrappers (e.g. maybe `electron-builder` when building the Android app of an Electron codebase?), or for test reports that are also in the JUnit format but for tests other than Unit tests (e.g. UI Tests / E2E Tests?).
If it happens that those other reports, despite being in the JUnit format, don't fit the assumption that this `merge_junit_reports` utility makes of them not being wrapped within `<testsuites>` root node because they are generated by another tool than Android Gradle Plugin for Unit Tests, then this will break the merged file that it generates.

By contrast, just adjusting the `sed` pattern to remove those `<testsuites>` root elements if they happen to be present in the input file doesn't seem like something too complex to add (iirc I even suggested a pattern for it in the WCAndroid PR where you introduced that script initially) while making this `merge_junit_reports` utility more resilient and less likely to create invalid XML files.

Reply:
Gotcha 👍 I've added a logic to ignore `<testsuites>` tag in reports that are being merged. WDYT? 1913f43ae9ceecbef370414154264f0f95373539

I've tried to add a test for this with `bats`, but surprisingly it happened to be quite unstable even with a simple test - occasionally it was ending in a state, that was simply not finishing a test. After killing the process and running test again all worked fine. I didn't want to introduce flakiness though.